core/permissions/permission_store.go

```diff
@@ -0,0 +1,57 @@
+package permissions
+
+import (
+	_ "embed"
+
+	"github.com/casbin/casbin/v2"
+	"github.com/casbin/casbin/v2/model"
+	"github.com/casbin/casbin/v2/persist"
+)
+
+type PermissionStore interface {
+	HasPermission(
+		didstr string,
```

Comment:
IIRC indigo has types for all of these. Not sure if we want to use them or not

Reply:
yeah we should use official types for these and i also wonder if instead of `write bool` we should have `permission permissionType` and that can be `write | read` and potentially more ?